packages/react/src/next-ssr-plugin.tsx

```diff
@@ -18,7 +19,7 @@ export function NextSSRPlugin(props: { routerConfig: EndpointMetadata }) {
   useServerInsertedHTML(() => {
     const html = [
       // Hydrate routerConfig on client globalThis
-      `globalThis.__UPLOADTHING = ${JSON.stringify(props.routerConfig)};`,
+      `globalThis.__UPLOADTHING = ${JSON.stringify(props.routerConfig, safeNumberReplacer)};`,
```

Comment:
i think this is redundant given both the GET endpoint and the `extractRouterCongi` (which is what you're meant to pass as props here) use `buildPermissionHandler` which in turn invokes `fillRouterInput...` which does the replacer - so it should be all valid JSON

Reply:
You're right, I just forgot about it 😅